### PR TITLE
Update OpenTelemetry SDK to the latest version (1.2.0-rc1) 

### PR DESCRIPTION
**Pull request by pichlermarc**
This PR updates the OTel SDK version to the latest version (1.2.0-rc1). The new SDK version introduced several breaking changes changes which made previously accessible methods inaccessible.

Main points include the introduction of a `TestMetricExporterProxy` to replace the `TestMetricReader` and changes in the conversion for Histograms.

Special thanks to @joaopgrassi for already doing some work beforehand :)

### CODE REVIEW

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/Dynatrace.OpenTelemetry.Exporter.Metrics.csproj [hunk 1]
```diff
@@ -18,7 +18,7 @@
   </PropertyGroup>
```
Comment (pirgeo) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics/Dynatrace.OpenTelemetry.Exporter.Metrics.csproj:18]: Please bump the package version above.

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs [hunk 2]
```diff
@@ -27,64 +27,75 @@ internal static class DynatraceMetricsExtensions
...
-				// In this case, only one bucket exists: (-Inf, Inf). If there were any boundaries, there
-				// would be more counts.
-				if (pointData.BucketCounts[0] > 0)
+				buckets.Add(bucket);
+			}
+
+			return buckets;
+		}
+
+		private static double GetMinFromBoundaries(double sum, long count, IReadOnlyList<HistogramBucket> buckets)
```
Comment (pirgeo) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs:82]: `count` is the number of values, right?

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs [hunk 3]
```diff
@@ -101,74 +112,70 @@ private static double GetMinFromBoundaries(MetricPoint pointData)
 						// - The average in the bucket (smallest if there are multiple positive measurements
 						// smaller than the lowest boundary)
 						return Math.Min(
-							Math.Min(pointData.ExplicitBounds[i], pointData.DoubleValue),
-							pointData.DoubleValue / pointData.LongValue);
+							Math.Min(buckets[i].ExplicitBound, sum),
```
Comment (pirgeo) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs:115]: I think i remember why we needed this average here: If we would just take the sum of all values smaller than the lowest value, we might actually get a value that is bigger than the smallest bound (e.g. smallest bound is 1, and we record 3 measurements of 0.6, the sum would be 1.8, and using that over 1 would make no sense in this case)

**Review by pirgeo — APPROVED** [review]
Looking good to me after version bump. Thanks for fixing the histogram code :)

**Review by joaopgrassi — APPROVED** [review]

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs (hunk 2 above)
Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs]: Correct, I changed the names to `histogramSum` and `histogramValueCount` to better reflect that. :)

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/Dynatrace.OpenTelemetry.Exporter.Metrics.csproj (hunk 1 above)
Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics/Dynatrace.OpenTelemetry.Exporter.Metrics.csproj:18]: bumped to `0.4.0-beta` :)

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs (hunk 3 above)
Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs]: Ah yes, that makes sense of course!

**Pull request merged**